default.py

```diff
@@ -25,139 +23,309 @@
-        episodes = reversed(episodes)
-    return control.draw_items(episodes)
-
-@route('newest')
-def NEWEST(payload):
-    return control.draw_items(_BROWSER.get_newest())
-
-@route('newest/*')
-def NEWEST_PAGES(payload):
-    return control.draw_items(_BROWSER.get_newest(int(payload)))
-
-@route('latest')
-def LATEST(payload):
-    return control.draw_items(_BROWSER.get_latest())
-
-@route('latest/*')
-def LATEST_PAGES(payload):
-    return control.draw_items(_BROWSER.get_latest(int(payload)))
-
-@route('recent_subbed')
-def SUBBED(payload):
-    return control.draw_items(_BROWSER.get_recent_subbed())
-
-@route('recent_subbed/*')
-def SUBBED_PAGES(payload):
-    return control.draw_items(_BROWSER.get_recent_subbed(int(payload)))
-
-@route('recent_dubbed')
-def DUBBED(payload):
-    return control.draw_items(_BROWSER.get_recent_dubbed())
-
-@route('recent_dubbed/*')
-def DUBBED_PAGES(payload):
-    return control.draw_items(_BROWSER.get_recent_dubbed(int(payload)))
-
-@route('popular_subbed')
-def POPSUBBED(payload):
-    return control.draw_items(_BROWSER.get_popular_subbed())
-
-@route('popular_subbed/*')
-def POPSUBBED_PAGES(payload):
-    return control.draw_items(_BROWSER.get_popular_subbed(int(payload)))
-
-@route('popular_dubbed')
-def POPDUBBED(payload):
-    return control.draw_items(_BROWSER.get_popular_dubbed())
-
-@route('popular_dubbed/*')
-def POPDUBBED_PAGES(payload):
-    return control.draw_items(_BROWSER.get_popular_dubbed(int(payload)))
-
-@route('search')
-def SEARCH(payload):
-    query = control.keyboard(control.lang(30007))
-    if query:
-        return control.draw_items(_BROWSER.search_site(query))
-    return False
-
-@route('search/*')
-def SEARCH_PAGES(payload):
-    query, page = payload.rsplit("/", 1)
-    return control.draw_items(_BROWSER.search_site(query,
-                                                            int(page)))
-
-@route('genres')
-def LIST_GENRES(payload):
-    return control.draw_items(_BROWSER.get_genres())
-
-@route('genre/*')
-def GENRE_ANIMES(payload):
-    genre, page = payload.rsplit("/", 1)
-    return control.draw_items(_BROWSER.get_genre(genre, int(page)))
-
-@route('play/*')
-def PLAY(payload):
-    anime_url, episode = payload.rsplit("/", 1)
-    sources = _BROWSER.get_episode_sources(anime_url, int(episode))
-
-    serverChoice = filter(lambda x:
-        control.getSetting(x[0]) == 'true', SERVER_CHOICES.iteritems())
-    serverChoice = map(lambda x: x[1], serverChoice)
-    sources = filter(lambda x: x[0] in serverChoice, sources)
+    prefereResSetting = utils.parse_resolution_of_source(plugin.get_setting('prefres'))
+    filtered_urls = filter(lambda x: utils.parse_resolution_of_source(x[0]) <= prefereResSetting, fetched_urls)
+    return sorted(filtered_urls, key=lambda x: utils.parse_resolution_of_source(x[0]), reverse=True)
+
 
-    autoplay = True if 'true' in control.getSetting('autoplay') else False
+MENU_ITEMS = [
+    ("Latest", "latest_d"),
```

Comment:
also here, you removed support or string table..

Reply:
I kind of didn't understand how to implement it with new framework. Will try to solve it.